[10.0]  backport stock_storage_type from 13.0

lmignon opened the pull request:
depends of
* [ ] https://github.com/OCA/server-tools/pull/1940
* [ ] https://github.com/OCA/stock-logistics-workflow/pull/744
* [ ] https://github.com/OCA/stock-logistics-workflow/pull/743
* [ ] https://github.com/OCA/product-attribute/pull/695
* [ ] https://github.com/OCA/product-attribute/pull/693

simahawk committed:
Add shopfloor_base (split from shopfloor)

simahawk committed:
shopfloor_base: add shopfloor.scenario

simahawk committed:
shopfloor_base: decouple actions lookup from service

simahawk committed:
shopfloor_base: 'schema' is an action as well

simahawk committed:
shopfloor_base: scan anything delegate search

Search and information retrieval is not hardcoded anymore.
You can register specific handlers from outside
to extend its behavior.

simahawk committed:
shopfloor_base: data|schema_detail are actions as well

simahawk committed:
shopfloor_base: actions_for must be private

The new api will try to load info for public methods on services.
Prevent it as having them public seems not relevant.

simahawk committed:
shopfloor_base: misc fixups

simahawk committed:
shopfloor_base: reorganize misc test classes

simahawk committed:
shopfloor_base: rework scan_anything tests

simahawk committed:
shopfloor: refactor on top of shopfloor_base

simahawk committed:
shopfloor: refactor scan_anything w/ components

simahawk committed:
shopfloor_batch_automatic_creation: refactor on shopfloor_base

simahawk committed:
shopfloor_checkout_sync: adapt to shopfloor_base

simahawk committed:
shopfloor: remove harcoded conf for scenario

Options can now be configured on scenario
hence is useless to have hardcoded behavior on menu model.

Commit:
shopfloor: add dependency on stock_helper

To remove the duplicate implementation of
StockLocation.is_sublocation_of()

Commit:
stock_dynamic_routing: add dependency on stock_helper

To remove the duplicate implementation of
StockLocation.is_sublocation_of()

Commit:
stock_move_source_relocate: add dependency on stock_helper

To remove the duplicate implementation of
StockLocation.is_sublocation_of()

simahawk committed:
shopfloor_base: scan_anything allow limit record types scan

simahawk committed:
shopfloor_mobile: split base

simahawk committed:
shopfloor_mobile: fix menu

simahawk committed:
shopfloor: checkout disable_no_package compat w/ yaml options

lmignon committed:
stock_storage_type: putaway strategy "do not mix" should give priority to location that already contains the product (the one with the less qty)
fixes #168

santostelmo committed:
[13.0][IMP] Show warning in reception screen without web_notify dependency

simahawk committed:
shopfloor_mobile: depend on shopfloor_base

simahawk committed:
shopfloor: fix rebase on 13.0

simahawk committed:
shopfloor: fix copyright

lmignon committed:
stock_storage_type: Optimisation of the sorting of valid locations

Co-authored-by: Jacques-Etienne Baudoux <je@bcim.be>

simahawk committed:
shopfloor_base: keep relation menu <-> scenario for existing records

simahawk committed:
shopfloor_packing_info: fix menu parent

simahawk committed:
shopfloor_workstation: fix menu parent

lmignon committed:
stock_storage_type: Finalisation of the optimisation of the sorting  of valid locations

simahawk committed:
shopfloor_mobile: fix relative import paths

simahawk committed:
shopfloor_mobile: fix assets path

simahawk committed:
shopfloor_mobile: main.css split from base

simahawk committed:
shopfloor: update descriptions

simahawk committed:
shopfloor_base: compute scenario for inactive menu

simahawk committed:
shopfloor_base: move menu to app, add quick link to logs

simahawk committed:
shopfloor_mobile_base: reorganize registries

simahawk committed:
shopfloor_mobile_base: add page_registry

simahawk committed:
shopfloor_mobile_base: settings pages from registry

simahawk committed:
Disable ESLint for all shopfloor modules

simahawk committed:
Add shopfloor_workstation_mobile

simahawk committed:
shopfloor_workstation: fix demo data/test

simahawk committed:
shopfloor: fix after rebase on 13.0

OCA-git-bot committed:
shopfloor_mobile 13.0.1.12.0

OCA-git-bot committed:
[UPD] addons table in README.md

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: wms-13.0/wms-13.0-shopfloor
Translate-URL: https://translation.odoo-community.org/projects/wms-13-0/wms-13-0-shopfloor/

simahawk committed:
shopfloor_base: pure json for scenario options edit

simahawk committed:
shopfloor: update README

simahawk committed:
shopfloor_base: update README

simahawk committed:
Add shopfloor_rest_log

simahawk committed:
shopfloor_base: drop dependency on rest_log

simahawk committed:
shopfloor_base: re-license to LGPL

ibuioli committed:
Translated using Weblate (Spanish (Argentina))

Currently translated at 100.0% (210 of 210 strings)

Translation: wms-13.0/wms-13.0-shopfloor
Translate-URL: https://translation.odoo-community.org/projects/wms-13-0/wms-13-0-shopfloor/es_AR/

simahawk committed:
shopfloor_mobile_base: add explicit hook for new scenario

simahawk committed:
shopfloor_base: fix group menu by scenario

simahawk committed:
shopfloor_base: add TODO for scenario model

simahawk committed:
shopfloor: fix missing menu view override

simahawk committed:
shopfloor: fix menu.active field already in base

OCA-git-bot committed:
Merge PR #174 into 13.0

Signed-off-by simahawk

OCA-git-bot committed:
stock_available_to_promise_release 13.0.1.11.1

OCA-git-bot committed:
[UPD] addons table in README.md

santostelmo committed:
Reset warn on button save and next pack

simahawk committed:
shopfloor_base: log what done via hooks

simahawk committed:
shopfloor: grant permission to stock users on update

simahawk committed:
shopfloor: fix package level unlink

https://github.com/odoo/odoo/commit/b33e72d0bf027fb2c789b1b9476f7edf1a40b0a6
changed the behavior of package level deletion on move lines.

lmignon committed:
stock_storage_type: browse only once and avoid recordset union

simahawk committed:
shopfloor: improve _fill_stock_for_moves in tests

simahawk committed:
shopfloor: fix stock.move.extract_and_action_done

There's no need to call action assign on the brand new backorder
because all lines at that point will be already assigned.

OCA-git-bot committed:
Merge PR #169 into 13.0

Signed-off-by simahawk

OCA-git-bot committed:
[UPD] addons table in README.md

OCA-git-bot committed:
[UPD] README.rst

OCA-git-bot committed:
[ADD] icon.png

OCA-git-bot committed:
[ADD] setup.py

oca-travis committed:
[UPD] Update shopfloor.pot

oca-travis committed:
[UPD] Update shopfloor_base.pot

oca-travis committed:
[UPD] Update shopfloor_rest_log.pot

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: wms-13.0/wms-13.0-shopfloor
Translate-URL: https://translation.odoo-community.org/projects/wms-13-0/wms-13-0-shopfloor/

OCA-git-bot committed:
Merge PR #165 into 13.0

Signed-off-by simahawk

OCA-git-bot committed:
stock_move_source_relocate 13.0.1.1.0

OCA-git-bot committed:
shopfloor 13.0.4.1.0

OCA-git-bot committed:
stock_dynamic_routing 13.0.1.1.0

OCA-git-bot committed:
[UPD] addons table in README.md

oca-travis committed:
[UPD] Update stock_move_source_relocate.pot

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: wms-13.0/wms-13.0-stock_move_source_relocate
Translate-URL: https://translation.odoo-community.org/projects/wms-13-0/wms-13-0-stock_move_source_relocate/

OCA-git-bot committed:
Merge PR #172 into 13.0

Signed-off-by simahawk

OCA-git-bot committed:
stock_reception_screen 13.0.1.6.0

OCA-git-bot committed:
[UPD] addons table in README.md

oca-travis committed:
[UPD] Update stock_reception_screen.pot

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: wms-13.0/wms-13.0-stock_reception_screen
Translate-URL: https://translation.odoo-community.org/projects/wms-13-0/wms-13-0-stock_reception_screen/

ibuioli committed:
Translated using Weblate (Spanish (Argentina))

Currently translated at 100.0% (195 of 195 strings)

Translation: wms-13.0/wms-13.0-shopfloor
Translate-URL: https://translation.odoo-community.org/projects/wms-13-0/wms-13-0-shopfloor/es_AR/

ibuioli committed:
Translated using Weblate (Spanish (Argentina))

Currently translated at 100.0% (99 of 99 strings)

Translation: wms-13.0/wms-13.0-stock_reception_screen
Translate-URL: https://translation.odoo-community.org/projects/wms-13-0/wms-13-0-stock_reception_screen/es_AR/

simahawk committed:
shopfloor: search prevent query w/ no value + add tests

simahawk committed:
shopfloor: search location by name too

simahawk committed:
shopfloor: location data barcode fallback to name

sebalix committed:
shopfloor_base: use partner.display_name

simahawk committed:
Rename shopfloor_mobile_custom_example -> shopfloor_example

lmignon committed:
Merge remote-tracking branch 'oca/13.0' into 10.0-backport-stock-storage-type

simahawk committed:
shopfloor_example: add custom service

lmignon committed:
[FIX] stock_storage_type: Fix singleton error
Consider the max height and max weight if more than one quant into the package

simahawk committed:
shopfloor_base: give demo user shopfloor permission

simahawk committed:
shopfloor_base: add generic msg for rec not found

simahawk committed:
shopfloor_base: service add properties for schema

simahawk committed:
shopfloor_base: validator return None if not found

simahawk committed:
sf_mobile_base: add shortcuts odoo.get + odoo.post

simahawk committed:
sf_mobile_base: remove workstation pt1

simahawk committed:
shopfloor_rest_log: activate on install

simahawk committed:
sf_mobile_base: do not load menu if not profile is set

simahawk committed:
sf_mobile_base: allow route config via registry

simahawk committed:
sf_workstation_mobile: allow route config via registry

simahawk committed:
sf_mobile_base: router use meta.requiresAuth for auth check

simahawk committed:
sf_mobile_base: profile required via route config

OCA-git-bot committed:
Merge PR #179 into 13.0

Signed-off-by simahawk

OCA-git-bot committed:
shopfloor_base 13.0.1.1.0

OCA-git-bot committed:
[UPD] addons table in README.md

OCA-git-bot committed:
Merge PR #178 into 13.0

Signed-off-by simahawk

OCA-git-bot committed:
shopfloor 13.0.4.2.0

OCA-git-bot committed:
[UPD] addons table in README.md

simahawk committed:
sf_mobile_base: fix translation on login

simahawk committed:
sf_mobile_base: delegate nav lookup to root

simahawk committed:
sf_mobile_base: refactor stored config handling

* add config registry
* refactor special properties which are stored on root and in browser
storage

simahawk committed:
sf_mobile_base: add TODO for storage lib

simahawk committed:
sf_mobile_base: get rid of workstation left over

simahawk committed:
sf_workstation: fix ACL

simahawk committed:
sf_workstation_mobile: plug config and transl. via registry

simahawk committed:
sf_mobile_base: remove picking form left over

simahawk committed:
sf_mobile_base: use translations where required

OCA-git-bot committed:
Merge PR #186 into 13.0

Signed-off-by simahawk

OCA-git-bot committed:
shopfloor_rest_log 13.0.1.0.1

OCA-git-bot committed:
[UPD] addons table in README.md

Commit:
[IMP] stock_reception_screen: new test

Adds a new test to reproduce a bug that appears when,
having more than two moves, the last one is packed
using the button "Next Pack". To do this, the existing
tests had to be refactored to avoid duplicating code.

lmignon committed:
[IMP] stock_storage_type: Prevent concurrent update by not writing on stock.location if computed values are the same as the one already into the database

Commit:
[FIX] stock_reception_screen: last line received with Next Pack

Fixes a crash that appeared when there were several lines to
pick and the last one was packed with the button "Next Pack".
For the case in which no more moves were waiting, there was
an assertion that was activated.

simahawk committed:
sf_mobile_base: add utils registry

simahawk committed:
shopfloor: split wms utils

simahawk committed:
sf_mobile_base: add TODO split misc components

sebalix committed:
[FIX] s_storage_type: no storage type for delivery

There is no need to set a package storage type on delivery packages (we
are not storing them).
This avoid to trigger blocking constraints (like height).

simahawk committed:
shopfloor: drop dependency on rest_log

simahawk committed:
shopfloor: fix inheritance of shopfloor.search.action

simahawk committed:
sf_mobile_base: move wms components to sf_mobile

Commit:
[FIX] stock_dynamic_routing_reserve_rule: fix Reservation Rules

Button to access the reservation rules from a dynamic routing
crashed because of a bad name in a field, probably because of
an old refactoring.

simahawk committed:
shopfloor_base: add ACSONE to credits

OCA-git-bot committed:
Merge PR #187 into 13.0

Signed-off-by simahawk

OCA-git-bot committed:
stock_reception_screen 13.0.1.6.1

OCA-git-bot committed:
[UPD] addons table in README.md

simahawk committed:
sf_mobile_base: fix favicon path

OCA-git-bot committed:
Merge PR #188 into 13.0

Signed-off-by simahawk

OCA-git-bot committed:
stock_storage_type 13.0.1.10.1

OCA-git-bot committed:
[UPD] addons table in README.md

OCA-git-bot committed:
Merge PR #189 into 13.0

Signed-off-by simahawk

OCA-git-bot committed:
stock_dynamic_routing_reserve_rule 13.0.1.0.1

OCA-git-bot committed:
[UPD] addons table in README.md

simahawk committed:
shofloor: make menu picking type info not mandatory

simahawk committed:
shopfloor: isolate menu items in tests

simahawk committed:
sf_mobile_base: decouple nav item templates

simahawk committed:
sf_mobile: override menu item templates

simahawk committed:
sf_mobile: wms_utils fix parseInt usage

simahawk committed:
sf_mobile: update ROADMAP

simahawk committed:
sf_mobile_base: add some TODOs

sebalix committed:
shopfloor: checkout, allows delivery packages only

A delivery package is a package with a delivery packaging defined
('packaging_id' field). It is a package different than internal ones
(which are stored in the warehouse with their own constraints like
height...).
Here we enforce the use of such delivery packages when we perform a
checkout, this means that we can not use internal packages to perform a
delivery.

* 'list_dest_package' method ('/select_dest_package') is now listing only
  delivery packages
* 'scan_dest_package' and 'set_dest_package' methods accept only a
  delivery package
* 'scan_package_action' method accept only a delivery package

simahawk committed:
sf_mobile: improve pkg qty picker display control

* use: options.non_zero_only to hide pakaging w/ no qty
* use: options.pkg_name_key to control which field is used for pkg name

simahawk committed:
sf_mobile: checkout display qty using qty widget

simahawk committed:
sf_mobile: checkout fix package line select handling

simahawk committed:
sf_mobile: keep manual select box always on top

simahawk committed:
shopfloor: checkout fix list existing packages

When listing available existing packages to select
display only destination packages.

simahawk committed:
shopfloor_mobile: fix usage of parseInt

simahawk committed:
shopfloor: bump 13.0.4.3.0

simahawk committed:
shopfloor_workstation: bump 13.0.1.1.1

OCA-git-bot committed:
Merge PR #190 into 13.0

Signed-off-by simahawk

OCA-git-bot committed:
[UPD] addons table in README.md

OCA-git-bot committed:
[ADD] setup.py

OCA-git-bot committed:
[UPD] README.rst

oca-travis committed:
[UPD] Update shopfloor_base.pot

oca-travis committed:
[UPD] Update shopfloor_example.pot

oca-travis committed:
[UPD] Update shopfloor_mobile.pot

oca-travis committed:
[UPD] Update shopfloor_mobile_base.pot

oca-travis committed:
[UPD] Update shopfloor_workstation_mobile.pot

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: wms-13.0/wms-13.0-shopfloor_mobile
Translate-URL: https://translation.odoo-community.org/projects/wms-13-0/wms-13-0-shopfloor_mobile/

sebalix committed:
shopfloor: checkout, select the delivery packaging

Add a new state/screen when the user click on the "New package" button
to list available delivery packaging ('select_delivery_packaging' state)
for the current transfer and select one to create the package (existing
'new_package' method which now accepts a packaging as parameter).

OCA-git-bot committed:
Merge PR #193 into 13.0

Signed-off-by simahawk

OCA-git-bot committed:
shopfloor 13.0.4.3.1

OCA-git-bot committed:
shopfloor_mobile 13.0.2.0.1

OCA-git-bot committed:
[UPD] addons table in README.md

Commit:
[IMP] stock_storage_type: allow to archive storage types/locations

* Usual field added.
* Simple tests added, to keep the coverage.
* Filter added to show the archived records.

simahawk committed:
shopfloor: delivery picking done only if moves are done

Before this change the check for done qty was done only on move lines.
Now the check is at move level, so that if there's any move not fully satisfied
the picking is not set to done automatically and you have to confirm.

lmignon committed:
Merge remote-tracking branch 'oca/13.0' into 10.0-backport-stock-storage-type

OCA-git-bot committed:
Merge PR #171 into 13.0

Signed-off-by simahawk

OCA-git-bot committed:
stock_storage_type 13.0.1.11.0

OCA-git-bot committed:
[UPD] addons table in README.md

OCA-git-bot committed:
Merge PR #192 into 13.0

Signed-off-by simahawk

lmignon committed:
Merge remote-tracking branch 'oca/13.0' into 10.0-backport-stock-storage-type

OCA-git-bot committed:
shopfloor_mobile_base 13.0.1.9.0

OCA-git-bot committed:
shopfloor_mobile 13.0.2.1.0

OCA-git-bot committed:
[UPD] addons table in README.md

OCA-git-bot committed:
Merge PR #196 into 13.0

Signed-off-by simahawk

OCA-git-bot committed:
shopfloor 13.0.4.3.2

OCA-git-bot committed:
[UPD] addons table in README.md

oca-travis committed:
[UPD] Update stock_storage_type.pot

simahawk committed:
sf_mobile: fix single pack transf confirm location change

Ensure that if we scan again the location
it gets confirmed without asking anymore.

simahawk committed:
shopfloor: single pack transf. ignore draft pkg level

When scanning a package, the list of related package levels are found.
This list can include records from draft picking that were left behind
which still reference the same package.
If this happens we don't care about them.

simahawk committed:
sf_mobile_base: state update trigger 'screen:reload'

sebalix committed:
shopfloor: checkout, select line after canceling

Once a line or package has been canceled, return on the `select_line`
state instead of `summary`.
When we cancel a line or a package, there is a high chance that the user
wants to restart again the operation.

simahawk committed:
sf_mobile_base: searchbar preserve focus on state refresh

When clicking on a button in a scenario state
is not granted the whole screen gets refreshed
and that the input field of the searchbar
gets its focus back.
Make sure it's always preserved.

OCA-git-bot committed:
Merge PR #198 into 13.0

Signed-off-by simahawk

OCA-git-bot committed:
shopfloor_mobile_base 13.0.1.9.1

OCA-git-bot committed:
[UPD] addons table in README.md

OCA-git-bot committed:
Merge PR #194 into 13.0

Signed-off-by simahawk

OCA-git-bot committed:
shopfloor 13.0.4.4.0

OCA-git-bot committed:
shopfloor_mobile 13.0.2.2.0

OCA-git-bot committed:
[UPD] addons table in README.md

OCA-git-bot committed:
Merge PR #195 into 13.0

Signed-off-by simahawk

OCA-git-bot committed:
stock_storage_type 13.0.1.12.0

OCA-git-bot committed:
[UPD] addons table in README.md

oca-travis committed:
[UPD] Update shopfloor.pot

oca-travis committed:
[UPD] Update stock_storage_type.pot

OCA-git-bot committed:
Merge PR #197 into 13.0

Signed-off-by simahawk

OCA-git-bot committed:
shopfloor 13.0.4.4.1

OCA-git-bot committed:
shopfloor_mobile 13.0.2.2.1

OCA-git-bot committed:
[UPD] addons table in README.md

simahawk committed:
sf_mobile_base: prevent scan multiple times

The scanning interface uses a simple input text
which is valued and submitted by the scanning device automatically.
While sending the request the focus stays on the input
hence if the user scans multiple times the same barcode
multiple requests are sent to the backend.

This change prevents this by checking if the app
is still waiting for data from the backend.

By doing so, several conflict errors are avoided
because the same action cannot be applied twice or more
for the same endpoint till the UI is unlocked.

OCA-git-bot committed:
Merge PR #199 into 13.0

Signed-off-by simahawk

OCA-git-bot committed:
shopfloor_mobile_base 13.0.1.9.2

OCA-git-bot committed:
[UPD] addons table in README.md

lmignon committed:
Merge remote-tracking branch 'oca/13.0' into 10.0-backport-stock-storage-type

lmignon committed:
Makes shopfloor_mobile_base and shopfloor_example not installable after backport

lmignon committed:
[IMP] stock_storage_type: Filter out locations to not recompute cache before trying to detect changes into the cached values

lmignon committed:
[FIX] stock_storage_type: Fix views after backport from 13.0
widget web_ribbon doesn't exit in 10.0

jbaudoux committed:
Change sorting of locations in the pack putaway computation

Invert the sorting for abc packs storage strategy to first apply the
max_height which is a physical constraint, then apply the abc
preference.

Add a pack putaway sequence to be able to favor some locations (e.g.
level). For ordered children strategy, finalize by sorting by location
name. For abc, random selection is applied at last as it was already.

Replaced multi-steps python sorting by an unique SQL quicksort query to
quickly compute the result on large stock_location table. This is
especially important for operations containing many moves requiring a
putaway computation.

jbaudoux committed:
fixup! Change sorting of locations in the pack putaway computation

lmignon committed:
Merge pull request #1 from jbaudoux/pr95-backport-13_fix_abc_maxheight

Change sorting of locations in the pack putaway computation

lmignon committed:
[FIX] tock_storage_type: Fix tests since parent_store is not always recomputed into test phase...
Strange and annoying error...

lmignon committed:
[IMP] stock_storage_type: Improve performance when recomputating the stock.location cache

Before this change, the cache was recomputed also for locations outside of the current scope due to the prefetch mechanism into odoo

jbaudoux committed:
Fix Package storage type message computation.

The message was broken with the changes in commit 6b25b867c2f606d5f4220915ab9aae3b30c1f87f

jbaudoux committed:
stock_storage_type: add location smart buttons

lmignon committed:
Merge pull request #3 from jbaudoux/pr95-fixmsg

Fix Package storage type message computation.

lmignon committed:
Merge pull request #4 from jbaudoux/pr95-locationssmartbuttons

stock_storage_type: add location smart buttons

jbaudoux committed:
[FIX] stock_storage_type: Fix putaway compution when a strategy is defined

Fix commit c4080a944237435c86be606a33d6c6cd1d9a921c

jbaudoux committed:
[FIX] stock_storage_type: Fix putaway compution when a strategy is defined

When the strategy has to find a leaf location, do not return non-leaf
locations defined on a storage type putaway sequence.

Example: sequence is defined to first look in WH/Stock/area1 and then in
WH/Stock/area2. On area1 the strategy is to find a leaf location under
area1. If area1 currently has no valid leaf location for the pack storage
type, area1 should not be returned as a valid leaf location.

jbaudoux reviewed (approved):
This PR works fine.
I have backported a last fix done on v13.0: https://github.com/acsone/wms/pull/5

Can be merged

lmignon committed:
Merge pull request #5 from jbaudoux/pr95-leaf_location

[FIX] stock_storage_type: Fix putaway compution when a strategy is defined

lmignon committed:
Merge pull request #6 from jbaudoux/pr95-leaf_location-fix

[FIX] stock_storage_type: Fix putaway compution when a strategy is defined

jbaudoux committed:
Fix stock_storage_type putaway sequence "show locations" button.

Fix backport error in commit 8518d7285c8793a5e9b143c6cad044c0ba3cbd90

lmignon committed:
Merge pull request #7 from jbaudoux/pr95-locationssmartbuttons-fix

Fix stock_storage_type putaway sequence "show locations" button.

lmignon committed:
[IMP] stock_storage_type: Be more aggressive on stock.location cache invalidation for computed fields

lmignon committed:
[IMP] stock_storage_type: Allow to restrict some putaway sequence

fixes #343

lmignon committed:
[FIX] stock_storage_type: adapt backport of 'Allow to restrict some putaway sequence'

commit 769f97cc6ab221908a6fb9bca2a44f7cdb745ba7

lmignon committed:
[IMP] stock_storage_type: Remove useless code

The pack_putaway_strategy==None and the validation of the sequence location is also done into the code of the method 'get_storage_locations'. This code could therefore be removed with the benefit of processing the strategy into the same place

lmignon committed:
[IMP] stock_storage_type: Add support for fixed location put away

If the put-away strategy is None, we also need to compute the odoo standard put-away on that destination.

refs #343

jbaudoux committed:
[FIX] stock_storage_type: fix location_is_empty computation

There could be pack operations with a qty_done linked to a move that is
not reserved anymore. Those should not be taken into account.

lmignon committed:
Merge pull request #13 from jbaudoux/10.0-backport-stock-storage-type-fix_location_empty

[FIX] stock_storage_type: fix location_is_empty computation
